### PR TITLE
[Agency Dashboards] Wrap CategoryOverview and DashboardView within the <Protected> component to enforce the allowlist

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

[Follow up for #883]

This wraps the `CategoryOverview` and the `DashboardView` within the `<Protected>` component to enforce the allowlist incase a user goes to the `/agency/:slug/:category` or `/agency/:slug/dashboard` paths directly. 

In this demo - you'll see an example of not wrapping the components and then one with wrapping them:

https://github.com/Recidiviz/justice-counts/assets/59492998/66316de7-5c6b-4acf-b252-2df0afa8a93b

Thankfully, I assumed wrong about the multiple fetch calls - since this component stays mounted, it will only ever call when you load any one of those wrapped components once and anytime the slug-line changes.

**Here's how I tested this**:
* Added a console.log to the `fetchAgencyData` method in the AgencyDataStore
* Go to the `/agency/:slug` path which loads the `AgencyOverview` component
* Confirm there is a console.log statement (our `fetchAgencyData` method ran)
* From the `AgencyOverview` go to the `/agency/:slug/:category` path (which loads the `CategoryOverview` component) by clicking on the **Understand the Finances ->** link
* Confirm that there is **_no_** additional console.log statement
* Go to the `/agency/:slug/:category` path by entering it directly in the URL which loads the `CategoryOverview` component (this is a fresh request)
* Confirm there is a console.log statement (our `fetchAgencyData` method ran)
* Click on the **<- Home** link which takes us to the `/agency/:slug` path which loads the `AgencyOverview` component
* Confirm that there is **_no_** additional console.log statement

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: agency-dashboard/src/CategoryOverview/CategoryOverview.tsx [hunk 1]
```diff
@@ -121,18 +119,6 @@ export const CategoryOverview = observer(() => {
...
-    try {
-      await agencyDataStore.fetchAgencyData(slug);
-    } catch {
-      showToast({
-        message: "Error fetching data.",
-        color: "red",
-        timeout: 4000,
-      });
-    }
-  }, []);
```
Comment (mxosman) [inline agency-dashboard/src/CategoryOverview/CategoryOverview.tsx:134]: No longer needed - this happens in the <Protected> component.

Comment (alfonsogoberjr) [reply agency-dashboard/src/CategoryOverview/CategoryOverview.tsx:134]: Super clean

**Review by alfonsogoberjr — APPROVED** [review]
LGTM

**Review by lilidworkin — APPROVED** [review]

**Comment by mxosman** [comment]
Thank you so much, Lili!

**Pull request merged**